### DISCUSSION TITLE
multiple instruments approach MR geneated an error

### DISCUSSION THREAD

**Discussion by mocksu** [MRCIEU/TwoSampleMR#300] [Q&A] [answered]
I have 8 SNPs shared between the instrument dataset and the outcome dataset. All SNPs in each dataset has rsID, beta, se, p-value, effect_allele, other_allele. The outcome dataset also has eaf while the instrument dataset does not have eaf. None of the SNPs have sample size.

With the two datasets, "harmonise_data(..)" passes all 8 SNPs -- all 8 SNPs remain in the dataset returned (let's call it "dat"). When I run "mr(dat)", it returns 5 rows, with one each for "MR Egger", "Weighted median", "Inverse variance weighted", "Simple mode", and "Weighted mode". 

However, if I run "mr(dat, method_list = c("mr_raps"))", I get the following error:

> Analysing 'jZJqkI' on 'QLoCPt'
> Error in mr.raps::mr.raps(data, diagnostics = FALSE, over.dispersion = parameters$over.dispersion,  : 
>   unused arguments (diagnostics = FALSE, shrinkage = parameters$shrinkage)
> 

Can someone please help?

Thanks so much in advance!

**Comment by mocksu** [comment, marked as answer]
> mr(dat, method_list = c("mr_ivw"))


worked. Also, just 

> mr(dat)


worked which automatically includes all multivariable methods and generate results for all. "mr_raps" is just buggy, I guess.

**Comment by linjf15** [comment]
I met the same question, too. Then I read the original codes for mr.raps::mr.raps, as shown below:
`mr.raps <- function (b_exp, b_out, se_exp, se_out, over.dispersion = FALSE, 
    loss.function = c("l2", "huber", "tukey"), 
    diagnosis = FALSE, se.method = c("sandwich", "bootstrap"), 
    k = switch(loss.function[1], l2 = NULL, huber = 1.345, tukey = 4.685), 
    B = 1000, suppress.warning = FALSE)
{...} `
Now the question is clear, there are no parameters named as diagnostics and shrinkage.
So you might need to rewrite the codes of TwoSampleMR:mr locally to amend this bug.

**Comment by 634765911** [comment]
I have same question. Do you fix this error? Can you help me ? Thank you！

**Reply by mocksu** [reply]
see my answer to 1475737557 below

**Reply by hagithub-haha** [reply]
Do you fix this error? Can you help me ? Thank you！

**Reply by hagithub-haha** [reply]
Do you fix this error? Can you help me ? Thank you！

**Comment by 1475737557** [comment]
I also have same question. Have you solved this problem yet? Can you share it with me? Thank you very much!

**Reply by mocksu** [reply]
As stated above, just run:

    mr(dat)

worked which automatically includes all multivariable methods and generate results for all.

**Reply by 634765911** [reply]
just run: 
mr(dat).  
It works well.
when I use method list includes "mr_raps", it works error. why?

**Reply by linjf15** [reply]
The parameters deliver to mr.raps::mr.raps by mr(method_list = "mr_raps") are not correct as I mentioned above on Jan 15, 2022. Personally, I made a little change to the "mr" function, see https://github.com/linjf15/MR_tricks/blob/main/main_analysis/mr_modified.R.

**Reply by wangyiyisheng** [reply]
Hi Jianfeng
              Thanks your mr_modified.
